src/transformers/models/llama/modeling_llama.py

```diff
@@ -1274,7 +1274,11 @@ def prepare_inputs_for_generation(
 
         # TODO @gante we should only keep a `cache_position` in generate, and do +=1.
         # same goes for position ids. Could also help with continued generation.
-        cache_position = torch.arange(past_length, past_length + position_ids.shape[-1], device=position_ids.device)
+        cache_position = (
```

Comment:
the attention mask is always passed to the model so the position ids are always created before this!

Reply:
If so, should we set `attention_mask` as a required parameter? like 
```python
def prepare_inputs_for_generation(
        self, input_ids, attention_mask, past_key_values=None, inputs_embeds=None, **kwargs
    ):
```

Because we cannot control user behavior, I think we should avoid this error in codes or set a warning.

Reply:
have not seen any issue so far. If past key values is not None, the attention mask is created as well and always passed to the model by the generate function. Let's just check if positions ids exist or just use input_ids instead for device placement and shape as it always exists

Reply:
Great idea! Have fixed it according to your comments:)